src/oops/SOLID/singleResponsibilityPrinciple/before/repository/EmployeePersistanceManager.java

```diff
@@ -0,0 +1,20 @@
+package oops.SOLID.singleResponsibilityPrinciple.before.repository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import oops.SOLID.singleResponsibilityPrinciple.before.model.Employee;
+import oops.SOLID.singleResponsibilityPrinciple.before.util.EmployeeSerializer;
+
+public class EmployeePersistanceManager {
+	
+	public void saveToFile(Employee employee) throws IOException {		
+		String seralizedEmployee = EmployeeSerializer.serialize(employee);
```

Comment:
In case if serialization logic changes, you will have to either change EmployeeSerializer class (which is violation of open-close principle) or create new serializer and then refer it in this class (violation of single responsibility). Better alternative will be to create a non static serializer, and pass object of that serializer in this class and use it as black box here

Reply:
The reason I thought of leaving it as a static method is that it seemed like a utility method to me.  I agree with you comment regarding the possibility of changing the serialization logic but if the requirement were to change, we would have to touch the EmployeeSerializer class anyway. What do you think?

Reply:
I can rather create an interface `Serializer<T, S>` and make `EmployeePersistanceManager` accept `Serializer<Employee, String>` either as constructor argument or method argument.

Next, I can make `EmployeeSerializer` to implement `Serializer<Employee, String>` and pass it to `EmployeePersistanceManager` on runtime. If in case requirements change later, I dont have to touch existing classes, but only create another class implementing `Serializer<Employee, String>` and pass it to `EmployeePersistanceManager` on runtime instead of `EmployeeSerializer`.

Hope I could explain it well.

If not, maybe you can check [this](https://github.com/AmeyKamat/low-level-design-concepts/tree/master/src/oops/SOLID/singleResponsibilityPrinciple/before) for the idea.

Reply:
cool. I like your idea. Thanks for the detailed explanation !!

Reply:
Happy to help :)

One more thing that catches my eye is the name of your persistence layer (`EmployeePersistanceManager`)
I remember vividly one senior dev suggesting me not to name classes generically (eg. Manager or Controller), since by looking at name of the class I am not able to guess the scope of operation of that class (managing persistence? what services does this class provide to higher layers by managing persistence?).

Not a hard and fast rule as such, but steer clear of naming something as Manager wherever possible.

Reply:
okay. I initially thought about naming a new class to specify 'File' somewhere, but I thought I might be creating too many classes for a small functionality. But after reading your explanation and going through your PR, I realized that it's better to name the classes better from the beginning to make things clear. I'll also remember to name them properly going forward. Thanks